docs/user-guide/src/bmo/instance_customization.md

```diff
@@ -0,0 +1,145 @@
+
+1. Per-host images take a lot of disk space, especially since Ironic has a
+   local image cache.
+2. *diskimage-builder* allows only basic customization out of box, code will
+   need to be written for anything complex.
+
+It is recommended to use [UserData](#userdata) or [NetworkData](#networkdata)
+instead when possible.
+
+## NetworkData
+
+*Network data* describes the desired networking configuration in the [OpenStack
+network_data.json][network_data] format supported by *cloud-init* and *Glean*.
+The format is not very well documented, but you can consult the [network_data
+JSON schema][network_data schema] shipped with OpenStack.
+
+Usually, one network data secret is created per host and should be linked to
+it. For example, given a local file `host-0-network.json`, you can create a
+secret:
+
+```bash
+kubectl create secret generic host-0-networkdata --from-file=networkData=host-0-network.json
+```
+
+Then you can attach it to the host during its enrollment or when starting
+provisioning:
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: host-0
+  namespace: my-cluster
+spec:
+  online: true
+  bootMACAddress: 80:c1:6e:7a:e8:10
+  bmc:
+    address: ipmi://192.168.1.13
+    credentialsName: host-0-bmc
+  image:
+    checksum: http://192.168.0.150/SHA256SUMS
+    url: http://192.168.0.150/jammy-server-cloudimg-amd64.img
+  networkData:
+    name: host-0-networkdata
+```
+
+[network_data]: https://docs.openstack.org/nova/latest/user/metadata.html#openstack-format-metadata
+[network_data schema]: https://docs.openstack.org/nova/latest/_downloads/9119ca7ac90aa2990e762c08baea3a36/network_data.json
+
+## UserData
+
+*User data* describes the desired configuration of the instance in a format
+specific to the first-boot software:
+
+* *cloud-init* supports two [formats][cloud-config]: *cloud-config* YAML and
+  a shell script (distinguished by the header).
+* *Ignition* uses its own [format][ignition-config].
+* *Glean* does not support user data at all.
+
+For example, you can create a *cloud-config* file `host-0.yaml`:
+
+```yaml
+#cloud-config
+users:
+- name: metal3
+  ssh_authorized_keys:
+  - ssh-ed25519 ABCD... metal3@example.com
+```
+
+```bash
+kubectl create secret generic host-0-userdata --from-file=userData=host-0.json
+```
+
+Then you can attach it to the host during its enrollment or when starting
+provisioning:
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: host-0
+  namespace: my-cluster
+spec:
+  online: true
+  bootMACAddress: 80:c1:6e:7a:e8:10
+  bmc:
+    address: ipmi://192.168.1.13
+    credentialsName: host-0-bmc
+  image:
+    checksum: http://192.168.0.150/SHA256SUMS
+    url: http://192.168.0.150/jammy-server-cloudimg-amd64.img
+  userData:
+    name: host-0-userdata
+```
+
+[cloud-config]: https://cloudinit.readthedocs.io/en/latest/explanation/format.html
+[ignition-config]: https://coreos.github.io/ignition/specs/
+
+## Implementation notes
+
```

Comment:
I think here it would be nice to expand the explanation of the config drive with the info that  BMO will combine the user, network and metadata  and pass it over to Ironic where Ironic will share it with IPA via the IPA API thus the config drive is not present right away when an IPA instance boots.

Reply:
Thanks for the doc btw, it is a nice addition and will help a lot of folks!